### DISCUSSION TITLE
Shipping pre-built assets?

### DISCUSSION THREAD

**Discussion by fauno** [ElMassimo/jekyll-vite#12] [Ideas]
Currently jekyll-vite always builds assets either in development or production modes.  This means all JS packages need to be installed during build, even if the source JS hasn't changed.

Would you be open to patches to statically distribute pre-built assets?  This would also enable using jekyll-vite while developing [Jekyll gem-based themes](http://jekyllrb.com/docs/themes/#understanding-gem-based-themes).

For this to work:

* `publicOutputDir: "assets"` should be default for production mode. Jekyll themes load assets from an `assets/` dir. Not sure yet where to configure jekyll-vite so it doesn't output files into `assets/assets/`
* Files should be saved to the Jekyll source dir instead of output dir

**Comment by ElMassimo** [comment]
The goal of the current integration is to improve the experience of developing a site, didn't consider themes because I personally never used themes for jekyll.

> Would you be open to patches to statically distribute pre-built assets?

It would be interesting to have a configuration mode that makes it easier to do that, but I wouldn't want that to be the default.

I'd like to hear more about the specific use cases you have in mind, without prescribing a solution.

**Reply by fauno** [reply]
Hi!  Thanks for replying. I have two use-cases in mind:

For gem-based Jekyll themes, all includes and layouts are packaged in a gem.  Assets are included if they're stored under an `assets/` dir, anything else is excluded.  You can have a `_posts/` dir and other static files for development, but they won't be distributed with the gem.  Even if you do, `Jekyll::Theme` won't load them.

So you need to bundle any assets beforehand, and jekyll-vite can't automate that at the moment.

The other use-case, which brought me here, is more related to our workflow at Sutty (but similar to gem-based release cycles really) is that when you're editing content and publishing your site, JS assets should be already bundled.  Having a whole bundling env around just to confirm no JS was modified feels like a waste of resources.  If you're building in a CI, you need a cache for `node_modules/` or reinstall packages every time, and stuff is moved around anyway.

What we do currently with Webpack 4 to bundle for production is to store the `manifest.json` inside `_data/`, and assets into `assets/js/`.  This way we can link to the latest bundle during each Jekyll build without installing Webpack or having it around.  These two steps are currently run at different moments in different machines.
